tests/Payments.Tests/ControllerTests/InvoiceControllerTests.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Internal;
+using Moq;
+using Payments.Core;
+using Payments.Controllers;
+using Payments.Core.Models;
+using Payments.Tests.Helpers;
+using Xunit;
+
+namespace Payments.Tests.ControllerTests
+{
+    public class InvoiceControllerTests
+    {
+        //private readonly Mock<PaymentsContext> _context;
+        //private readonly Mock<IMapper> _mapper;
+
+        //public InvoiceControllerTests()
+        //{
+
+        //   _context = new Mock<PaymentsContext>();
+        //   _mapper = new Mock<IMapper>();
+        //}
+
+        public class MyDbContext : DbContext
+        {
+            public MyDbContext()
+            {
+            }
+
+            public MyDbContext(DbContextOptions options)
+                : base(options)
+            {
+             
+            }
+
+
+        }
+
+        [Fact]
+        public void Test1()
+        {
+            var data = new List<Invoice>
+            {
+                CreateValidEntities.Invoice(1),
+                CreateValidEntities.Invoice(2),
+                CreateValidEntities.Invoice(3),
+            }.AsQueryable();
+
+
+
+            //var optionsBuilder = new DbContextOptionsBuilder<MyDbContext>();
+            var mockSet = MockInvoice(data);
+            var mockContext = new Mock<PaymentsContext>();
+            mockContext.Setup(m => m.Invoices).Returns(mockSet.Object);
+
+
+
+            var mapper = new Mock<IMapper>(); //Probably don't want to mock this. Maybe assign it directly?
+            var controller = new InvoiceController(mockContext.Object, mapper.Object);
+
+            var result = controller.Index();
+
+            var viewResult = Assert.IsType<ViewResult>(result);
+            var model = Assert.IsType<Invoice[]>(viewResult.Model);
+            Assert.Equal(3, model.Count());
+
+        }
+
+        private static Mock<DbSet<Invoice>> MockInvoice(IQueryable<Invoice> data)

```

Comment:
@jSylvestre This would be really easy to do with a generic and even as an extension method and it would really clean up the tests.  private static `Mock<DbSet<T>> MockSet<T>()`